eng/source-build-pre-release.yml

```diff
@@ -0,0 +1,146 @@
+        branch_version=$(echo $sdk_version | sed 's/..$/xx/')
+        branch_name="refs/heads/internal/release/${branch_version}"
+
+        commit=$(jq -r --arg BRANCH "${branch_name}" '.builds[] | select(.repo == "https://dev.azure.com/dnceng/internal/_git/dotnet-installer") | select(.branch == $BRANCH) | .commit' $manifest_path)
+
+        if [[ ! $commit ]]; then
+          echo "Installer commit not set. Exiting..."
+          exit 1
+        fi
+
+        echo "Release channel: ${release_channel}"
+        echo "Runtime version: ${runtime_version}"
+        echo "SDK version: ${sdk_version}"
+        echo "Installer branch: ${branch_name}"
+        echo "Installer commit: ${commit}"
+
+        echo "##vso[task.setvariable variable=SdkVersion;isOutput=true]${sdk_version}"
+        echo "##vso[task.setvariable variable=RuntimeVersion;isOutput=true]${runtime_version}"
+        echo "##vso[task.setvariable variable=ReleaseChannel;isOutput=true]${release_channel}"
+        echo "##vso[task.setvariable variable=InstallerCommit;isOutput=true]${commit}"
+      name: ReadReleaseInfoStep
+      displayName: Read Release Info
+
+- stage: UploadSourceTarballStage
+  displayName: Upload Source Tarball
+  dependsOn: GetBuildInfoStage
+  jobs:
+  - job: UploadSourceTarballJob
+    displayName: Upload Source Tarball
+    variables:
+    - name: InstallerCommit
+      value: $[ stageDependencies.GetBuildInfoStage.GetBuildInfoJob.outputs['ReadReleaseInfoStep.InstallerCommit'] ]
+    - name: SdkVersion
+      value: $[ stageDependencies.GetBuildInfoStage.GetBuildInfoJob.outputs['ReadReleaseInfoStep.SdkVersion'] ]
+    - name: RuntimeVersion
+      value: $[ stageDependencies.GetBuildInfoStage.GetBuildInfoJob.outputs['ReadReleaseInfoStep.RuntimeVersion'] ]
+    - name: ReleaseChannel
+      value: $[ stageDependencies.GetBuildInfoStage.GetBuildInfoJob.outputs['ReadReleaseInfoStep.ReleaseChannel'] ]
+    steps:
+    - script: |
+        set -euxo pipefail
+        echo $(ReleaseChannel)
+        echo $(RuntimeVersion)
+        echo $(SdkVersion)
+        echo $(InstallerCommit)
+      name: TestVariablesStep
+      displayName: Test Variables
+    - script: |
+        set -euo pipefail
+        az config set extension.use_dynamic_install=yes_without_prompt
+
+        installer_official_runs=$(az pipelines runs list --organization $(azDoOrg) --project $(azDoProject) --pipeline-ids $(installerOfficialCiPipelineId))
+        installer_official_run_id=$(echo $installer_official_runs | jq -r '.[] | select(.sourceVersion == "$(InstallerCommit)") | .id')
+        echo "dotnet-installer-official-ci run ID: ${installer_official_run_id}"
+
+        installer_tarball_build_runs=$(az pipelines runs list --organization $(azDoOrg) --project $(azDoProject) --pipeline-ids $(installerTarabllBuildCiPipelineId))
+        installer_tarball_build_run_id=$(echo $installer_tarball_build_runs | jq -r '.[] | select(.sourceVersion == "$(InstallerCommit)") | .id')
```

Comment:
How does this behave if there are multiple builds for the same commit?

Reply:
It looks like it will output both IDs.

```bash
> json='[{"id": 1, "commit": "sha1"}, {"id": 2, "commit": "sha2"}, {"id": 3, "commit": "sha2"}]'
> echo $json | jq '.[] | select(.commit == "sha1") | .id'
1
> echo $json | jq '.[] | select(.commit == "sha2") | .id'
2
3
> result=$(echo $json | jq '.[] | select(.commit == "sha2") | .id')
> echo $result
2 3
```

I wonder what we should do in this situation. Select the most recent? Select the most recent that succeeded?

Reply:
This is also a valid question for what we would do manually.

Reply:
Manually, I would expect a human could deduct why two builds existed.  One thing I would suggest is making sure the build passed.  I would say a fallback could be to have a variable to pass in the id.  If specified use the supplied build otherwise fail if there are multiple passing builds.

Reply:
In my latest change I try to detect if there are more than one build per commit and fail with an error if so. I also added the parameters for optional specific build IDs.